src/Storages/StorageDictionary.cpp

```diff
@@ -45,6 +47,11 @@ StorageDictionary::StorageDictionary(
     }
 }
 
+void StorageDictionary::checkTableCanBeDropped() const
+{
+    throw Exception("Cannot detach dictionary " + backQuoteIfNeed(dictionary_name) + " as table, use DETACH DICTIONARY query.", ErrorCodes::UNKNOWN_TABLE);
+}
```

Comment:
@azat Hello. I've just tried new build `ClickHouse server version 20.4.5.36 (official build).` and I can't delete Table with Engine=Dictionary. Always have a exception to use DETACH DICTIONARY when I try to remove whole database...

I think here can be an error. Can you help me?